core/src/test/java/nl/nn/adapterframework/pipes/CsvParserPipeTest.java

```diff
@@ -70,6 +70,77 @@ public void testFieldSeparator() throws Exception {
 		PipeRunResult prr = doPipe(csv);
 		assertXmlEquals(expected,prr.getResult().asString());
 	}
+
+	@Test
+	public void testHeaderCaseUpper() throws Exception {
+		pipe.setFieldNames("p,q,r");
+		pipe.setHeaderCase("uppercase");
+		pipe.setFileContainsHeader(true);
+		configureAndStartPipe();
+		String csv ="a,b,c\n1,2,3\nx,\"y,y\"";
+		String expected="<csv><record><P>1</P><Q>2</Q><R>3</R></record><record><P>x</P><Q>y,y</Q></record></csv>";
+		
+		PipeRunResult prr = doPipe(csv);
+		assertXmlEquals(expected,prr.getResult().asString());
+	}
+
+	@Test
+	public void testHeaderCaseLower() throws Exception {
+		pipe.setFieldNames("P,q,R");
+		pipe.setHeaderCase("lowercase");
+		pipe.setFileContainsHeader(true);
+		configureAndStartPipe();
+		String csv ="a,b,c\n1,2,3\nx,\"y,y\"";
+		String expected="<csv><record><p>1</p><q>2</q><r>3</r></record><record><p>x</p><q>y,y</q></record></csv>";
+		
+		PipeRunResult prr = doPipe(csv);
+		assertXmlEquals(expected,prr.getResult().asString());
+	}
+
+	@Test
+	public void testHeaderCaseNotSpecified() throws Exception {
+		pipe.setFieldNames("p,q,R");
+		pipe.setFileContainsHeader(true);
+		configureAndStartPipe();
+		String csv ="a,b,c\n1,2,3\nx,\"y,y\"";
+		String expected="<csv><record><p>1</p><q>2</q><R>3</R></record><record><p>x</p><q>y,y</q></record></csv>";
+		
+		PipeRunResult prr = doPipe(csv);
+		assertXmlEquals(expected,prr.getResult().asString());
+	}
+
+	@Test
+	public void testHeaderFromFileCaseLower() throws Exception {
+		pipe.setHeaderCase("lowercase");
+		pipe.setFileContainsHeader(true);
+		configureAndStartPipe();
+		String csv ="A,b,c\n1,2,3\nx,\"y,y\"";
+		String expected="<csv><record><A>1</A><b>2</b><c>3</c></record><record><A>x</A><b>y,y</b></record></csv>";
+		
+		PipeRunResult prr = doPipe(csv);
+		assertXmlEquals(expected,prr.getResult().asString());
+	}
 	
+	@Test
+	public void testNoHeaderFromFileCaseLower() throws Exception {
+		pipe.setHeaderCase("lowercase");
+		configureAndStartPipe();
+		String csv ="A,b,c\n1,2,3\nx,\"y,y\"";
+		String expected="<csv><record><A>1</A><b>2</b><c>3</c></record><record><A>x</A><b>y,y</b></record></csv>";
+		
+		PipeRunResult prr = doPipe(csv);
+		assertXmlEquals(expected,prr.getResult().asString());
+	}
 
+	@Test
+	public void testUseFieldNameAsHeaderFileContainesHeaderFalse() throws Exception {
```

Comment:
Could you also add a test for FileContainsHeader=false without fieldNames specified?

Reply:
this test case exists, see line 34